harness/determined/experimental/__init__.py

```diff
@@ -1,14 +1,20 @@
-from determined.common.experimental import (
-    Checkpoint,
-    Determined,
-    ExperimentReference,
-    Model,
-    ModelOrderBy,
-    ModelSortBy,
-    TrialReference,
-    TrialOrderBy,
-    TrialSortBy,
-)
+import warnings
+
+with warnings.catch_warnings(record=True):
```

Comment:
I don't see that on my warnings module? The __init__ on the version I have is as such:
```python3
class catch_warnings(object):

    """A context manager that copies and restores the warnings filter upon
    exiting the context.

    The 'record' argument specifies whether warnings should be captured by a
    custom implementation of warnings.showwarning() and be appended to a list
    returned by the context manager. Otherwise None is returned by the context
    manager. The objects appended to the list are arguments whose attributes
    mirror the arguments to showwarning().

    The 'module' argument is to specify an alternative module to the module
    named 'warnings' and imported under that name. This argument is only useful
    when testing the warnings module itself.

    """

    def __init__(self, *, record=False, module=None):
        """Specify whether to record warnings and if an alternative module
        should be used other than sys.modules['warnings'].

        For compatibility with Python 3.0, please consider all arguments to be
        keyword-only.

        """
        self._record = record
        self._module = sys.modules['warnings'] if module is None else module
        self._entered = False
```

When I try:
```python3
with warnings.catch_warnings(category=FutureWarning, record=True):
```
 I get this error:
```
>>> from determined.experimental import Checkpoint
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/taylor/git/determined/harness/determined/experimental/__init__.py", line 3, in <module>
    with warnings.catch_warnings(category=FutureWarning, record=True):
TypeError: __init__() got an unexpected keyword argument 'category'
```

I like the idea but I'm not seeing how it could actually be used

Reply:
Aw, dang.

> Changed in version 3.11: Added the action, category, lineno, and append parameters.